dashboard/app/controllers/api/v1/ml_models_controller.rb

```diff
@@ -27,6 +29,13 @@ def user_ml_model_names
     render json: user_ml_model_data.to_json
   end
 
+  # GET api/v1/ml_models/metadata/:model_id
+  # Retrieve a trained ML model's metadata
```

Comment:
Instead of an additional controller action, what if we returned metadata in the get model names method so we don't need to make an additional ajax request?  I guess if a user had a bunch of models it could be big/slow, but it's just a few additional text fields per model, right? Then the dialog would have everything it needed to render regardless of which model is selected with just one server hit. Only on clicking import would we do the heavy lifting of calling for the actual big, trained model.

Reply:
Good point; I'll track this for a potential followup.  I did consider that approach up front, but worried that in the future there could be some giant responses for very active users, which we've had happen with some of our other APIs that can have unbounded numbers of results.